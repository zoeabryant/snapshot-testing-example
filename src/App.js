import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import Apple from './Apple';
import Orange from './Orange';
import Tea from './Tea';

class App extends Component {
  render() {
    const {likesApple, likesJuice} = this.props;
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <h1 className="App-title">Welcome to React</h1>
        </header>
        <div className="App-intro">
          {likesApple && likesJuice && <Apple />}
          {!likesApple && likesJuice && <Orange />}
          {!likesJuice && <Tea />}
        </div>
      </div>
    );
  }
}

export default App;
