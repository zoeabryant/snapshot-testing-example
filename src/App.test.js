import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import App from './App';

it('renders without crashing', () => {
  const tree = mount(<App />);
  expect(toJson(tree)).toMatchSnapshot();
});


it('renders Apple if user likes Apple and Juice', () => {
    const tree = shallow(<App likesApple likesJuice />);
  expect(toJson(tree)).toMatchSnapshot();
})

it('renders Orange if user likes Juice but not Apple', () => {
    const tree = shallow(<App likesJuice />);
  expect(toJson(tree)).toMatchSnapshot();
})

it('renders Tea if user does not like Juice, even though they like Apple', () => {
    const tree = shallow(<App likesApple likesJuice={false} />);
  expect(toJson(tree)).toMatchSnapshot();
})
