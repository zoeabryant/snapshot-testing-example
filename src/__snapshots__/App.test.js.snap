// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Apple if user likes Apple and Juice 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="App-intro"
  >
    <Apple />
  </div>
</div>
`;

exports[`renders Orange if user likes Juice but not Apple 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="App-intro"
  >
    <Orange />
  </div>
</div>
`;

exports[`renders Tea if user does not like Juice, even though they like Apple 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
  </header>
  <div
    className="App-intro"
  >
    <Tea />
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <h1
        className="App-title"
      >
        Welcome to React
      </h1>
    </header>
    <div
      className="App-intro"
    >
      <Tea>
        <h2>
          Have Tea!
        </h2>
      </Tea>
    </div>
  </div>
</App>
`;
